common/src/main/java/com/datastrato/gravitino/json/JsonUtils.java

```diff
@@ -933,4 +967,102 @@ public Expression deserialize(JsonParser p, DeserializationContext ctxt) throws
       return readFunctionArg(node);
     }
   }
+
+  public static class PartitionDTOSerializer extends JsonSerializer<PartitionDTO> {
+    @Override
+    public void serialize(PartitionDTO value, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeStartObject();
+      gen.writeStringField(PARTITION_TYPE, value.type().name().toLowerCase());
+      gen.writeStringField(PARTITION_NAME, value.name());
+      switch (value.type()) {
+        case IDENTITY:
+          IdentityPartitionDTO identityPartitionDTO = (IdentityPartitionDTO) value;
+          gen.writeFieldName(FIELD_NAMES);
+          gen.writeObject(identityPartitionDTO.fieldNames());
+          gen.writeArrayFieldStart(IDENTITY_PARTITION_VALUES);
+          for (LiteralDTO literal : identityPartitionDTO.values()) {
+            writeFunctionArg(literal, gen);
+          }
+          gen.writeEndArray();
+          break;
+        case LIST:
+          ListPartitionDTO listPartitionDTO = (ListPartitionDTO) value;
+          gen.writeArrayFieldStart(LIST_PARTITION_LISTS);
+          for (LiteralDTO[] literals : listPartitionDTO.lists()) {
+            gen.writeStartArray();
+            for (LiteralDTO literal : literals) {
+              writeFunctionArg(literal, gen);
+            }
+            gen.writeEndArray();
+          }
+          gen.writeEndArray();
+          break;
+        case RANGE:
+          RangePartitionDTO rangePartitionDTO = (RangePartitionDTO) value;
+          gen.writeFieldName(RANGE_PARTITION_UPPER);
+          writeFunctionArg(rangePartitionDTO.upper(), gen);
+          gen.writeFieldName(RANGE_PARTITION_LOWER);
+          writeFunctionArg(rangePartitionDTO.lower(), gen);
+          break;
+        default:
+          throw new IOException("Unknown partition type: " + value.type());
+      }
+      gen.writeObjectField(PARTITION_PROPERTIES, value.properties());
+      gen.writeEndObject();
+    }
+  }
+
+  public static class PartitionDTODeserializer extends JsonDeserializer<PartitionDTO> {
+    @Override
+    public PartitionDTO deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode node = p.getCodec().readTree(p);
+      Preconditions.checkArgument(
+          node != null && !node.isNull() && node.isObject(),
+          "Partition must be a valid JSON object, but found: %s",
+          node);
+      Preconditions.checkArgument(
+          node.has(PARTITION_TYPE), "Partition must have a type field, but found: %s", node);
+      String type = getString(PARTITION_TYPE, node);
+      switch (PartitionDTO.Type.valueOf(type.toUpperCase())) {
+        case IDENTITY:
+          List<String[]> fieldNames = Lists.newArrayList();
+          node.get(FIELD_NAMES).forEach(field -> fieldNames.add(getStringArray((ArrayNode) field)));
```

Comment:
Should you check it the "node.get(FIELD_NAMES)" is null?

Reply:
fixed